src/api/event/controllers/event.js

```diff
@@ -1,9 +1,38 @@
-'use strict';
+const { createCoreController } = require('@strapi/strapi').factories;
 
-/**
- * event controller
- */
+module.exports = createCoreController('api::event.event', ({ strapi }) => ({
+  async find(ctx) {
+    // filter with the query
+    const { entityId, entityType } = ctx.query;
 
-const { createCoreController } = require('@strapi/strapi').factories;
+    // Fetch only the events that match the filters
+    const filters = {};
+    if (entityId) filters.entityId = entityId;
+    if (entityType) filters.entityType = entityType;
+
+    const events = await strapi.entityService.findMany('api::event.event', {
+      filters,  // Apply filters to fetch specific events
+      populate: ['eventUser'],  // Populate relations (e.g., eventUser)
+    });
+
+    // Retrieve latest entityName for each event Dynamically
+    const latestEvents = await Promise.all(events.map(async (event) => {
```

Comment:
Hey Oriyomi, Promisse.all braks my mind a litle bit.
Do you know the advantage to use that here in this case ?

Reply:
The Promise.all allows us to run all of the processing in parallel. When we call `.map` the list is processed in sequence. Handling all of the events in the `.all()` allows us to run this process in parallel.